src/FSharp.Data.GraphQL.Server/Execution.fs

```diff
@@ -507,86 +494,109 @@ let private executeQueryOrMutation (resultSet: (string * ExecutionInfo) []) (ctx
                     k::kvps, e@errs) ([],[])
             return NameValueLookup(dicts |> List.rev |> List.toArray), (errors |> List.rev)
         }
-
+    let rec traversePath (d : DeferredExecutionInfo) (fieldCtx : ResolveFieldContext) (path: string list) (tree: AsyncVal<ResolverTree>) (pathAcc: string list): AsyncVal<(ResolverTree * string list) []> =
+        asyncVal {
+            let! tree' = tree
+            let! res = 
+                match List.tail path, tree' with
+                | [], t -> 
+                    asyncVal { 
+                        let! res = buildResolverTree d.Info.ReturnDef fieldCtx fieldExecuteMap t.Value
+                        match d.Info.Kind with
+                        | SelectFields [f] -> return! async { return [|res, List.rev (f.Identifier :: pathAcc)|] }
+                        | _ -> return! async { return [|res, List.rev ((List.head path)::pathAcc)|] }
+                    }
+                | [p], t -> 
+                    asyncVal { 
+                        let! res = buildResolverTree d.Info.ReturnDef fieldCtx fieldExecuteMap t.Value
+                        return! async { return [|res, List.rev(p::pathAcc)|] }
+                    }
+                | [p;"__index"], t ->
+                    asyncVal { 
+                        let! res = buildResolverTree d.Info.ReturnDef fieldCtx fieldExecuteMap t.Value
+                        return! async { return [|res, List.rev(p::pathAcc)|] }
+                    }
+                | [_;"__index";_;"__index"], _ ->
+                    raise <| GraphQLException("Nested defer directives are not supported yet.")
```

Comment:
We still need to treat this case, which is being handled incorrectly. For now I made a protection to tell the user that nested defer directives inside stream ones are not supported yet.